File: 1688-count-of-matches-in-tournament/1688-count-of-matches-in-tournament.py
class Solution(object):
    def numberOfMatches(self, n):
        """
        :type n: int
        :rtype: int
        """
        res = 0
        while n>1:
            x=n//2
            res+=x
            if n%2==0:
                n=x
            else:
                n=x+1
        return res

        

File: 1688-count-of-matches-in-tournament/1688-count-of-matches-in-tournament.cpp
class Solution {
public:
    int numberOfMatches(int n) {
        int res = 0;
        while(n>1){
            int x=n/2;
            res+=x;
            if(n%2==0){
                n=x;
            }
            else{
                n=x+1;
            }
        }
        return res;
    }
};